### PR TITLE
Add softmax_with_cross_entropy_op to ngraph engine

### PR DESCRIPTION
**Pull request by baojun-nervana**

### CODE REVIEW

**Review by mozga-intel — DISMISSED** [review]
LGTM!

**Comment by tensor-tang** [comment]
could you give some description whay is op_helper for? why only softmax use it?

**Comment by baojun-nervana** [comment]
I put some common shared function in the op_helper. Such as GetSoftmax can be used for softmax op build and softmax_with_cross_entropy op. It is to reduce some code duplicate.

**Comment by tensor-tang** [comment]
Did you mean will add other functions in op_helper? no only softmax?

**Comment by luotao1** [comment]
> Such as GetSoftmax can be used for softmax op build and softmax_with_cross_entropy op. 

I think `op_helper` is not fit for only softmax. How about include `softmax_op.h` in `cross_entropy.h`?

**Comment by baojun-nervana** [comment]
Removed op_helper.h. Functions moved to softmax_op.h and cross_entropy_op.h

**Review by tensor-tang — APPROVED** [review]
LGTM
